cmd/metrics/metrics.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const runtimeClassName = "kata-remote"
+
+var (
+	runtimeClassAvailable = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "kata_remote_runtimeclass_available",
+		Help: "Indicates if the " + runtimeClassName + " RuntimeClass is available (1) or not (0).",
+	})
+
+	kataConfigInstallationSuccess = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "kata_config_installation_success",
+		Help: "Indicates if KataConfig installation is successful (1) or not (0).",
+	})
+
+	kataRemoteWorkloadFailureRatio = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "kata_remote_workload_failure_ratio",
+		Help: "Percentage of kata-remote workloads that have failed.",
```

Comment:
Sorry  I didn't mention this one :

```suggestion
		Help: "Percentage of " + runtimeClassName + " workloads that have failed.",
```